### PR TITLE
Cherry picking Mojave fix (#30716) to release 2.1

### PR DESCRIPTION
**Pull request by maryamariyan**
Related to issue #30599 
Cherry picks PR #30716

cc: @danmosemsft @bartonjs

### CODE REVIEW

File: src/Native/Unix/System.Security.Cryptography.Native.Apple/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,9 @@ set(NATIVECRYPTO_SOURCES
     pal_x509chain.cpp
 )
 
+# Temporary workaround for dotnet/corefx issue #30599
+add_compile_options(-Wno-deprecated-declarations)
```
[the remark this answers is not in the document]
Comment (filipnavara) [reply src/Native/Unix/System.Security.Cryptography.Native.Apple/CMakeLists.txt]: It's weird you didn't need it. Did you also upgrade your Xcode? It happened to me with both Xcode 10 betas that were released so far.

[the remark this answers is not in the document]
Comment (filipnavara) [reply src/Native/Unix/System.Security.Cryptography.Native.Apple/CMakeLists.txt]: Yeah, the Xcode installation can be a bit of a pain if you want to keep different versions around...

```
xcode-select -p
/Applications/Xcode-beta.app/Contents/Developer
```

**Comment by danmosemsft** [comment]
@bartonjs for 2.1 servicing, do you want to take just this change, or do we have a reason to bring over https://github.com/dotnet/corefx/pull/30815 as well?

It seems that this change is sufficient to keep building and running on Mohave working, and it's less churn/risk than using the new API in servicing (that's real fix for master)

**Comment by bartonjs** [comment]
I'm fine with whichever fix for servicing.  I wouldn't expect Apple to be so aggressive as to the removal of the old function to cause us a problem during the support lifetime; and if we ever hit a problem we can take the other one.

**Comment by danmosemsft** [comment]
Ok let's take just this smaller fix to ship room. @maryamariyan Could you please add a row to the table again? I deleted the previous one while we waited...

**Pull request merged**